backend/src/packages/users/user.entity.ts

```diff
@@ -97,13 +97,15 @@ class UserEntity implements Entity {
     fullName: string;
     createdAt: Date;
     updatedAt: Date;
+    passwordHash: string;
   } {
     return {
       id: this.id as number,
       email: this.email,
       fullName: this.fullName,
       createdAt: this.createdAt as Date,
       updatedAt: this.updatedAt as Date,
+      passwordHash: this.passwordHash,
```

Comment:
Be careful not to send this field to the frontend when we request users

Reply:
yes, agree, let's remove passwordHash from toObject method and create a getter
```
  public get passwordData(): {
    passwordHash: string;
    passwordSalt: string;
  } {
    return {
      passwordSalt: this.passwordSalt as string,
      passwordHash: this.passwordHash as string,
    };
  }
```

Reply:
We can just do it from model layer via
```js
public static override get modifiers(): Modifiers<QueryBuilder<User>> {
    return {
      withoutPassword(builder): QueryBuilder<User> {
        return builder.select('id', 'email', 'createdAt', 'updatedAt');
      },
    };
  }
```

Reply:
I need passwordHash for password comparing, can I return user.toObject from auth.service that will be without passwordHash? Or can you clarify how I am supposed to use the code written by Volodymyr?

Reply:
Let's proceed with additional entity like UserWithPasswordEntity, so the default behaviour is to get user without password fields